### ISSUE TITLE
Realm Relationship Returning Empty Results

### ISSUE THREAD

**Issue by jimjeffers** [realm/realm-cocoa#1525]
Hey there,

I'm having unexpected behavior with a simple one-to-many relationship. I have an RLMObject called Item that has many ItemPhotos:
    
    let realm = RLMRealm.defaultRealm()
    let item = Item(forPrimaryKey: itemJSON["item_id"].intValue)
    var updatedPhotos: Array<ItemPhoto> = []
    for (index, photoJSON) in enumerate(itemJSON["item_photos"].arrayValue){
        let photo = ItemPhoto()
        photo.fileName = photoJSON["file_name"].stringValue
        photo.url = photoJSON["photo_url"].stringValue
        photo.item = item
        updatedPhotos.append(photo)
    }    
    realm.beginWriteTransaction()
    item.itemPhotos.addObjects(updatedPhotos)
    realm.commitWriteTransaction()
       
I'm adding the photos via `addObjects` and if I view the database in the RealmBrowser, everything looks fine. But if I try to fetch the item and get a count of its photos the count is 0.
 
     let item = Item.allObjects().firstObject() as Item
     let relationalCount = item.itemPhotos.count
     let queriedCount = ItemPhoto.objectsWhere("item = %@", item).count
     println("Queried Count: \(queriedCount) | Relationship Count: \(relationalCount)")

The output I get:

    Queried Count: 1 | Relationship Count: 0

If I perform a println() statement on the item's description I get:

    Item {
	  id = 693;
	  itemName = "test";
	  itemPhotos = RLMArray <0x7fdb0e01c1b0> (
    );

Any idea why the RLMArray would be empty even though Realm Browser and querying the items explicitly with a predicate show the results?

**Comment by segiddins** [comment]
@jimjeffers could you possibly share your `ItemPhoto` model definition? I have a hunch about what may be going on here.

**Comment by jimjeffers** [comment]
Sure, here's a simplified version of both models (just omitted extra props to keep this short):

Item:

    class Item: RLMObject {
        override class func primaryKey() -> String {
            return "id"
        }
        dynamic var id = 0
        dynamic var itemPhotos = RLMArray(objectClassName: ItemPhoto.className())
    }

Item photo:

    class ItemPhoto: RLMObject {
        override class func primaryKey() -> String {
            return "fileName"
        }
        dynamic var item: Item?
        dynamic var fileName = ""
    }

**Comment by segiddins** [comment]
@jimjeffers would it be possible for you to send over your Realm file or a more complete reproduction case? (You can email to help@realm.io if you'd prefer to keep things private). Thanks!

**Comment by jimjeffers** [comment]
@segiddins Thanks for following up on this. I actually solved the problem. I'm still new to Realm so I was using it improperly. I am basically using realm to cache objects from our API so all of this data ephemeral. My algorithm was destroying the item and rebuilding a new one from the cache before I deleted the child photo objects. So the child objects connected to the item persisted but the local item did not have references to them in their child array! Ah the joys of cache management...
